application/controllers/UserManagementController.php

```diff
@@ -1174,8 +1175,8 @@ public function actionBatchApplyRoles()
             $model = $this->loadModel($sItem);
             $aResults[$sItem]['title'] = $model->users_name;
 
-            //check if user is admin otherwhise change the role
-            if (intval($sItem) == 1) {  //todo REFACTORING is admin id always 1?? is there another possibility to check for admin user?
+            if (Permission::isForcedSuperAdmin($sItem)) {
```

Comment:
Is this part of the reported issue?
Code looks OK. I am just rising my hand as to make sure it is tested.

Reply:
No :part of `//todo REFACTORING is admin id always 1?? is there another possibility to check for admin user?`

isFoircedAdmin return https://github.com/LimeSurvey/LimeSurvey/blob/49eab4bf643943d3b02f6f81420a70af055c11d4/application/config/config-defaults.php#L673 (by default).

But superadmin 1 can be deleted or not be a superadmin in 3,5 or 6.

Reply:
there are a TODO, the function is here since years , the dev who add it don't know this function (we need a better place for such function)

Reply:
Nevermind...

Reply:
PS : i think this part can be removed … you can remove the superadmin permission to forcedsuperadmin : it stay a forcedsuperadmin :)

There are no difference if you don't update config.php settings here.

I can remove the fix if you prefer. But since there are a TODO with an existing solution done exactly for this : unsue .